app/src/main/java/com/doubtless/doubtless/screens/dashboard/usecases/DeleteAccountUseCase.kt

```diff
@@ -0,0 +1,57 @@
+package com.doubtless.doubtless.screens.dashboard.usecases
+
+import androidx.annotation.WorkerThread
+import com.doubtless.doubtless.constants.FirestoreCollection
+import com.doubtless.doubtless.screens.auth.usecases.UserManager
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.firestore.FirebaseFirestore
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.tasks.await
+import kotlinx.coroutines.withContext
+
+class DeleteAccountUseCase(
+    private val firestore: FirebaseFirestore,
+) {
+
+    sealed class Result {
+        class Success(val message: String) : Result()
+        class Error(val message: String) : Result()
+    }
+
+    @WorkerThread
+    suspend fun deleteAccount(userManager: UserManager): Result = withContext(Dispatchers.IO) {
+        try {
+
+//            Delete Doubts Posted by User
+            var db = firestore.collection(FirestoreCollection.AllDoubts)
+            var query =
+                db.whereEqualTo("author_id", userManager.getCachedUserData()!!.id).get().await()
```

Comment:
can't we delete all the document directly, like where().delete instead of first getting all the doubts then delete them

Reply:
There is no way to delete all the documents directly, we have to first retrieve all the documents and then delete them one by one.